src/main/java/com/autotune/analyzer/experiment/ExperimentInterfaceImpl.java

```diff
@@ -59,69 +62,99 @@ public boolean updateExperimentStatus(KruizeObject kruizeObject, AnalyzerConstan
+                    String type = resultK8sObject.getType();
+                    String namespace = resultK8sObject.getNamespace();
+
+                    K8sObject k8sObject = k8sObjectMap.getOrDefault(name, new K8sObject(name, type, namespace));
+                    HashMap<String, ContainerData> containerDataMap =
+                            k8sObject.getContainerDataMap() != null ? k8sObject.getContainerDataMap() : new HashMap<>();
+                    HashMap<String, NamespaceData> namespaceDataMap =
+                            k8sObject.getNamespaceDataMap() != null ? k8sObject.getNamespaceDataMap() : new HashMap<>();
+
+                    if (resultK8sObject.getContainerDataMap() != null && !resultK8sObject.getContainerDataMap().isEmpty()) {
+                        resultK8sObject.getContainerDataMap().forEach((cName, resultContainerData) -> {
+                            ContainerData containerData = containerDataMap.getOrDefault(
+                                    cName, new ContainerData(cName, resultContainerData.getContainer_image_name(),
+                                            resultContainerData.getContainerRecommendations(),
+                                            resultContainerData.getMetrics()));
+
+                            containerData.setResults(mergeResults(
+                                    containerData.getResults(),
+                                    resultContainerData.getResults(),
+                                    resultData.getIntervalStartTime(),
+                                    resultData.getIntervalEndTime()));
+
                             containerDataMap.put(cName, containerData);
-                        }
+                        });
                         k8sObject.setContainerDataMap(containerDataMap);
-                        k8sObjectHashMap.put(dName, k8sObject);
+                    } else if (resultK8sObject.getNamespaceDataMap() != null && !resultK8sObject.getNamespaceDataMap().isEmpty()) {
+                        resultK8sObject.getNamespaceDataMap().forEach((nsName, resultNamespaceData) -> {
+                            NamespaceData namespaceData = namespaceDataMap.getOrDefault(
+                                    nsName, new NamespaceData(nsName,
+                                            resultNamespaceData.getNamespaceRecommendations(),
+                                            resultNamespaceData.getMetrics()));
+
+                            namespaceData.setResults(mergeResults(
+                                    namespaceData.getResults(),
+                                    resultNamespaceData.getResults(),
+                                    resultData.getIntervalStartTime(),
+                                    resultData.getIntervalEndTime()));
+
+                            namespaceDataMap.put(nsName, namespaceData);
+                        });
+                        k8sObject.setNamespaceDataMap(namespaceDataMap);
                     }
```

Comment:
Would be good to catch any other non supported types and log it here.

Reply:
Handled in #1577